text/0117-typo-setting-api.md

```diff
@@ -0,0 +1,295 @@
+
+Meilisearch can adapt to different use-cases by providing customization options for the typo tolerance feature.
+
+## 3. Functional Specification
+
+### 3.1. `typo` API resource definition
+
+| Field                                            | Type            | Required |
+|--------------------------------------------------|-----------------|----------|
+| [enabled](#311-enabled)                          | Boolean         | False    |
+| [disabledAttributes](#312-disabledattributes)    | Array of String | False    |
+| [disabledWords](#313-disabledwords)              | Array of String | False    |
+| [minWordSizeForTypos](#314-minwordsizefortypos)  | Object          | False    |
+
+#### 3.1.1. `enabled`
+
+- Type: Boolean
+- Required: False
+- Default: `true`
+
+Whether the typo tolerance feature is enabled.
+
+##### 3.1.1.1. Impacts on the `typo` ranking rule
+
+The presence of `typo` in the ranking rules setting does not influence the activation/deactivation of the typo tolerance feature.
+
+If the `rankingRules` parameter of the index settings does not contain the `typo` rule, the results are not sorted according to the number of typos found.
+
+The typo tolerance feature is still used to match documents.
+
+### 3.1.2. `disabledAttributes`
+
+- Type: Array of String
+- Required: False
+- Default: `[]`
+
+`disabledAttributes` disable the typo tolerance feature on the specified attributes list.
+
+Any attributes defined in `disabledAttributes` won't have their values matched by the typo tolerance.
+
+#### 3.1.2.1. Example
+
+Given `["title"]` as `disabledAttributes` and the following document
+
+```json
+{
+    "id": 0,
+    "title": "Hey World"
+}
+```
+
+The engine won't try to match query term with typos with values of the `title` attribute at search time to match documents.
+
+- Typing `Warld` won't bring the document as a candidate for the search results.
+
+### 3.1.3. `disabledWords`
+
+- Type: Array of String
+- Required: False
+- Default: `[]`
+
+`disabledWords` disable the typo tolerance feature on a list of search query terms.
+
+> This field is case insensitive since the document attributes values are lowercased and de-unicoded internally.
+
+#### 3.1.3.1. Example
+
+If `Javascript` is specified in `disabledWords`, the engine won't apply the typo tolerance on the query term `Javascript` or `javascript` if typed at search time to match documents.
+
+### 3.1.4. `minWordSizeForTypos`
+
+- Type: Object
+- Required: False
+
+#### 3.1.4.1. `minWordSizeForTypos` Object Definition
+
+| Field                      | Type            | Required |
+|----------------------------|-----------------|----------|
+| [oneTypo](#3142-onetypo)   | Integer         | False    |
+| [twoTypos](#3143-twotypos) | Integer         | False    |
+
+#### 3.1.4.2. `oneTypo`
+
+- Type: Integer
+- Required: False
+- Default: `5`
```

Comment:
Is the default only when `enabled` is set to `true?`
Or does it apply even when typo tolerance is disabled?

Reply:
When typo tolerance is not enabled, the other properties are modifiable but this will have no impact on the search since typo tolerance is globally disabled.

So yes, `disabledAttributes`, `disabledWords`, `minWordSizeForTypos` fields will be taken into account at search when `enabled` is `true`.